lib/rubocop/cop/style/random_with_offset.rb

```diff
@@ -0,0 +1,179 @@
+      #   rand(6) + 1
+      #
+      #   # bad
+      #   1 + rand(6)
+      #
+      #   # bad
+      #   rand(6) - 1
+      #
+      #   # bad
+      #   1 - rand(6)
+      #
+      #   # bad
+      #   rand(6).succ
+      #
+      #   # bad
+      #   rand(6).pred
+      #
+      #   # bad
+      #   Random.rand(6) + 1
+      #
+      #   # bad
+      #   Kernel.rand(6) + 1
+      #
+      #   # bad
+      #   rand(0..5) + 1
+      #
+      #   # good
+      #   rand(1..6)
+      #
+      #   # good
+      #   rand(1...7)
+      class RandomWithOffset < Cop
+        MSG = 'Prefer ranges when generating random numbers instead of ' \
+          'integers with offsets.'.freeze
+
+        def_node_matcher :integer_op_rand?, <<-PATTERN
+          (send
+            int {:+ :-}
+            (send
+              {nil? (const nil? :Random) (const nil? :Kernel)}
+              :rand
+              {int irange erange}))
+        PATTERN
+
+        def_node_matcher :rand_op_integer?, <<-PATTERN
+          (send
+            (send
+              {nil? (const nil? :Random) (const nil? :Kernel)}
+              :rand
+              {int irange erange})
+            {:+ :-}
+            int)
+        PATTERN
+
+        def_node_matcher :rand_modified?, <<-PATTERN
+          (send
+            (send
+              {nil? (const nil? :Random) (const nil? :Kernel)}
+              :rand
+              {int irange erange})
+            {:succ :pred :next})
+        PATTERN
+
+        def on_send(node)
+          unless integer_op_rand?(node) || rand_op_integer?(node) ||
+                 rand_modified?(node)
+            return
+          end
+          add_offense(node)
+        end
+
+        private
+
+        def autocorrect(node)
+          lambda do |corrector|
+            if integer_op_rand?(node)
+              corrector.replace(node.source_range,
+                                corrected_integer_op_rand(node))
+            elsif rand_op_integer?(node)
+              corrector.replace(node.source_range,
+                                corrected_rand_op_integer(node))
+            elsif rand_modified?(node)
+              corrector.replace(node.source_range,
+                                corrected_rand_modified(node))
+            end
+          end
+        end
+
+        def corrected_integer_op_rand(node)
+          left, operator, right = node.children
+
+          offset = int_from_int_node(left)
+
+          prefix_node, _, random_node = right.children
+
+          prefix = prefix_from_prefix_node(prefix_node)
+          left_int, right_int = boundaries_from_random_node(random_node)
+
+          if operator == :+
+            "#{prefix}(#{offset + left_int}..#{offset + right_int})"
+          else
+            "#{prefix}(#{offset - right_int}..#{offset - left_int})"
+          end
+        end
+
+        def corrected_rand_op_integer(node)
+          left, operator, right = node.children
+
+          prefix_node, _, random_node = left.children
```

Comment:
prefix_node? :-)

Reply:
I meant the `Random.rand` or `Kernel.rand` etc. Can't think of a better name :)

Reply:
We generally call it `receiver`. 🙂